doc/source/developing/testing.rst

```diff
@@ -3,503 +3,400 @@
-      yt/frontends/artio/tests/test_outputs.py
-
-If the answer doesn't exist on the server yet, ``nosetests`` is run twice and
-during first pass ``--answer-store`` is added to the commandline.
-
 Updating Answers
 ~~~~~~~~~~~~~~~~
 
 In order to regenerate answers for a particular set of tests it is sufficient to
-change the answer name in *tests/tests.yaml* e.g.:
+change the ``answer_version`` attribute in the desired test class.
 
-.. code-block:: diff
-
-   --- a/tests/tests.yaml
-   +++ b/tests/tests.yaml
-   @@ -25,7 +25,7 @@
-        - yt/analysis_modules/halo_finding/tests/test_rockstar.py
-        - yt/frontends/owls_subfind/tests/test_outputs.py
-
-   -  local_owls_000:
-   +  local_owls_001:
-        - yt/frontends/owls/tests/test_outputs.py
-
-      local_pw_000:
-
-would regenerate answers for OWLS frontend.
-
-When adding tests to an existing set of answers (like ``local_owls_000`` or ``local_varia_000``),
+When adding tests to an existing set of answers (like ``local_owls_000.yaml`` or ``local_varia_000.yaml``),
 it is considered best practice to first submit a pull request adding the tests WITHOUT incrementing
 the version number. Then, allow the tests to run (resulting in "no old answer" errors for the missing
 answers). If no other failures are present, you can then increment the version number to regenerate
 the answers. This way, we can avoid accidentally covering up test breakages.
 
-Adding New Answer Tests
-~~~~~~~~~~~~~~~~~~~~~~~
+.. _handling_dependencies:
+
+Handling yt Dependencies
+------------------------
 
-In order to add a new set of answer tests, it is sufficient to extend the
-*answer_tests* list in *tests/tests.yaml* e.g.:
+We attempt to make yt compatible with a wide variety of upstream software
+versions. However, sometimes a specific version of a project that yt depends on
+causes some breakage and must be blacklisted in the tests or a more
+experimental project that yt depends on optionally might change sufficiently
+that the yt community decides not to support an old version of that project.
 
-.. code-block:: diff
+To handle cases like this, the versions of upstream software projects installed
+on the machines running the yt test suite are pinned to specific version
+numbers that must be updated manually. This prevents breaking the yt tests when
+a new version of an upstream dependency is released and allows us to manage
+updates in upstream projects at our pace.
 
-   --- a/tests/tests.yaml
-   +++ b/tests/tests.yaml
-   @@ -60,6 +60,10 @@
-        - yt/analysis_modules/absorption_spectrum/tests/test_absorption_spectrum.py:test_absorption_spectrum_non_cosmo
-        - yt/analysis_modules/absorption_spectrum/tests/test_absorption_spectrum.py:test_absorption_spectrum_cosmo
+If you would like to add a new dependency for yt (even an optional dependency)
+or would like to update a version of a yt dependency, you must edit the
+``tests/test_requirements.txt`` file, this path is relative to the root of the
```

Comment:
this is fine for now, but note that I'm planning to unify our requirement files into setup.cfg so unless that project is rejected I'll be updating this section later :)

Reply:
@jcoughlin11 this still needs to be adressed/answered

Reply:
@neutrinoceros Oh, sorry, I didn't think I needed to do anything since your work was in a different PR. I was thinking your changes would get added in after this PR was merged. Did I misunderstand?